src/models/payments.js

```diff
@@ -251,13 +245,21 @@ function isTraderFunds (config, funds) {
 }
 
 function * updateSourceTransfer (updatedTransfer, traderCredit, ledgers, config) {
-  const destinationTransfer = traderCredit.memo && traderCredit.memo.destination_transfer
-  if (!destinationTransfer) return
-  ledgers.validateTransfer(destinationTransfer)
-
-  const isTransferReady = updatedTransfer.state === 'prepared' || updatedTransfer.state === 'executed'
+  const isTransferReady = updatedTransfer.state === 'prepared' ||
+    (updatedTransfer.state === 'executed' && !updatedTransfer.execution_condition)
   if (!isTransferReady) return
 
+  if (!config.ledgerCredentials[updatedTransfer.ledger]) {
+    throw new AssetsNotTradedError('This connector does not support the given asset pair')
+  }
+
+  // TODO this is cheating, but how do we know the type of the final transfer's ledger?
```

Comment:
`five-bells-ledger` is a type of ledger. Bitcoin is another type of ledger. We want to the connector to be able to work with any type of ledger so this means that all ledger interactions should go through the [`multiledger` abstraction](https://github.com/interledger/five-bells-connector/blob/master/src/lib/ledgers/multiledger.js).

I think @sentientwaffle said that is cheating because we can't actually assume that a transfer on a different ledger will have the `ledger` field. I think we will need to have some kind of wrapper around transfers from other ledgers to make them easier to work with (i.e. `{ ledger: 'bitcoin', tx: '...' }` and maybe pull out some of the other fields as a convenience)